### ISSUE TITLE
Ensure all signin unblock flow events are reported

### ISSUE THREAD

**Issue by shane-tomlinson** [mozilla/fxa-auth-server#1505]
Ensure we have [flow events for](https://github.com/mozilla/fxa/pull/194/files#diff-c84b6b3195c5290ee85400577cc04f23R36):

* [ ] Attempt to login (`flow.signin.submit`)
* [ ] Had login requests blocked for security reasons? (`account.login.blocked`)
* [x] Had unblock codes sent? (`account.login.sentUnblockCode`)
* [x] Successfully verified an unblock code? (`account.login.confirmedUnblockCode`)
* [ ] Used an invalid unblock code? (`account.login.invalidUnblockCode`)

ref https://github.com/mozilla/fxa-auth-server/pull/1471#issuecomment-251552928
ref https://github.com/mozilla/fxa-content-server/pull/4154

**Comment by philbooth** [comment]
Fwiw, I think these should be flow events rather than activity events.

I just noticed that `account.login.sentUnblockCode` and `account.login.confirmedUnblockCode` are being emitted as activity events, which seems wrong to me. That's my fault for not explaining the distinction to you guys better and I'm going to PR a fix shortly, but just wanted to note it here so we don't make the same mistake again.

(although also, I'm part-way through a refactoring that renders this mistake impossible to make in the future)

**Comment by philbooth** [comment]
> I think these should be flow events rather than activity events.

Actually the second one, `Chose to verify their account to unblock themselves?`, probably is an activity event because it represents a state change on the account in the db. But the others definitely seem more flowish to me.

**Comment by rfk** [comment]
> Actually the second one, Chose to verify their account to unblock themselves?, probably is an activity event

Sounds like it should be covered by the existing `account.verified` activity event.

**Comment by shane-tomlinson** [comment]
> > Actually the second one, Chose to verify their account to unblock themselves?, probably is an activity event
> Sounds like it should be covered by the existing account.verified activity event.

I have updated the line to be `Successfully verified an unblock code? (account.login.confirmedUnblockCode)`

`account.verified` looks like it is only emit in `/recovery_email/verify_code`, which is slightly different to what we are doing here.

**Comment by shane-tomlinson** [comment]
@philbooth says for `Attempt to login (account.login.attempt ?)` we can use `flow.signin.submit` and `flow.force_auth.submit`, or more concretely `WHERE type LIKE 'flow.%.submit`.  More discussion on this particular metric needed.

**Comment by philbooth** [comment]
> @philbooth says for `Attempt to login (account.login.attempt ?)` we can use `flow.signin.submit` and `flow.force_auth.submit`, or more concretely `WHERE type LIKE 'flow.%.submit`.  More discussion on this particular metric needed.

Just to follow up that it was pointed out in IRC using `LIKE` won't work here because the requirement is to exclude sign-ups. So the query will need to AND and signin and force-auth. However, even that won't be correct because of the hateful, hateful sign-in-from-the-sign-up-view feature.

**Comment by philbooth** [comment]
> However, even that won't be correct because of the hateful, hateful sign-in-from-the-sign-up-view feature.

I am addressing this in my fix for #4300. The key is that we need to distinguish between the user clicking on the button and the content server sending a request, in order to properly model the current state of affairs.

So I'm keeping the `flow.${viewName}.submit` events as indicating a button press, and I'm adding `flow.${action}.attempt` (those names may change colour in the wash) to indicate a request to the auth server.

So in the case of signin-from-signup we would see:

1. `flow.signup.begin`
2. `flow.signup.engage`
3. `flow.signup.submit`
4. `flow.signin.attempt`

For this issue, that `flow.signin.attempt` would be the thing you want to count.

**Comment by seanmonstar** [comment]
Should we also include in the flow when an unblock code is tried, and is invalid? Invalid reasons would be 1) typo, 2) expired, 3) guessing/attacking. It's probably useful to see in a flow "the user tried 2 times with unblock codes, and gave up after the second time", ya?

**Comment by philbooth** [comment]
> Should we also include in the flow when an unblock code is tried, and is invalid?

Absolutely!

**Comment by seanmonstar** [comment]
Since rejecting an unblock code is something that a user would do as a 1-time action (they were going about their day, and see in their email some unblock code, and reject it), it's not really something that is part of a signin flow. In that sense, it doesn't make sense to be a flow event, so I'm killing that from the list.

There is currently a `log.info({ op: 'account.login.rejectedUnblockCode' })` in there, which means it should appear in the info logs in kibana. If we need it somewhere else, we can open a separate issue.

**Comment by rfk** [comment]
Related: https://github.com/mozilla/fxa-features/issues/48

**Comment by rfk** [comment]
(related to the rejecting of codes, I mean)
